frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -353,90 +384,72 @@ export const StorageClassEncryption: React.FC<ProvisionerProps> = ({
     namespace: CEPH_STORAGE_NAMESPACE,
     name: KMSConfigMapCSIName,
   };
-
   const [csiConfigMap, csiConfigMapLoaded, csiConfigMapLoadError] = useK8sWatchResource<
     ConfigMapKind
   >(csiCMWatchResource);
 
+  const setEncryptionId = React.useCallback(
+    (encryptionId: string) => {
+      setServiceName(encryptionId);
+      onParamChange(parameterKey, encryptionId, false);
+    },
+    [onParamChange, parameterKey],
+  );
+
+  /** When csiConfigMap is deleted from another tab "csiConfigMapLoadError" == true (404 Not Found),
+   * but, "csiConfigMap" still contains same old object that was present before the deletion of the configMap.
+   * Hence, dropdown was not updating dynamically. Used csiKmsDetails to handle that.
+   */
+  const [csiKmsDetails, setCsiKmsDetails] = React.useState<ConfigMapKind>(null);
   React.useEffect(() => {
-    if (isKmsSupported && !_.isEmpty(csiConfigMap)) {
-      const serviceNames = Object.keys(csiConfigMap?.data);
-      const kmsData = JSON.parse(csiConfigMap?.data[serviceNames[serviceNames.length - 1]]);
-      setCurrentKMS(kmsData);
-      const url = parseURL(kmsData.VAULT_ADDR);
-      const port = getPort(url);
-      const kmsObj: KMSConfig = {
-        name: {
-          value: kmsData.KMS_SERVICE_NAME,
-          valid: true,
-        },
-        address: {
-          value: `${url.protocol}//${url.hostname}`,
-          valid: true,
-        },
-        port: {
-          value: port,
-          valid: true,
-        },
-        backend: kmsData.VAULT_BACKEND_PATH,
-        caCert: state.kms.caCert ?? null,
-        caCertFile: state.kms.caCertFile,
-        tls: kmsData.VAULT_TLS_SERVER_NAME,
-        clientCert: state.kms.clientCert ?? null,
-        clientCertFile: state.kms.clientCertFile,
-        clientKey: state.kms.caCert ?? null,
-        clientKeyFile: state.kms.clientKeyFile,
-        providerNamespace: kmsData.VAULT_NAMESPACE,
-        hasHandled: true,
-      };
-      setEncryptionDispatch(SCActionType.SET_KMS_ENCRYPTION, '', dispatch, kmsObj);
-    } else {
-      setInProgress(false);
+    if (csiConfigMapLoaded && !csiConfigMapLoadError && csiConfigMap) {
+      setCsiKmsDetails(csiConfigMap);
+    }
+    // setting isExistingKms = false, when configMap not present (404 Not Found)
+    else if (csiConfigMapLoadError) {
+      setIsExistingKms(false);
+      setCsiKmsDetails(null);
+      setEncryptionId('');
```

Comment:
It's not possible when data is deleted csiConfigMap should be empty and better you use useDeepCompareMemoize after a watch

Reply:
use useDeepCompareMemoize after a watch --> IIUC useDeepCompareMemoize will not cause any difference here cause I am already passing the csiConfigMap object in the dependency array of useEffect.

It's not possible when data is deleted csiConfigMap should be empty --> yes, APi response do not contain any data, but, csiConfigMap still contains  the same old object that was present before the deletion of the configMap. Most probably this is an issue with useK8sWatchResource.

Reply:
The useEffect won't do deep comparision,  All it checks elements in its dependency array is changed or not. If the object is used as a dependency then it won't check properties of the objects are changed.  So for each render, you may get a different object with the same properties and useEffect will invoke every time. When we use watcher it is better you invoke deep compare, So if properties not changed, it will return the old object again.

Reply:
It's not possible when data is deleted csiConfigMap should be empty --> yes, APi response do not contain any data, but, csiConfigMap still contains the same old object that was present before the deletion of the configMap. Most probably this is an issue with useK8sWatchResource. -->  csiConfigMap not empty but csiConfigMapLoadError is not empty, if csiConfigMapLoadError found you are doing 

    setIsExistingKms(false);
    setCsiKmsDetails(null);
     setEncryptionId('');
   
   If setIsExistingKms is false then you are going to display an empty form only, then why you need this csiKmsDetails? you can stick with csiConfigMap, we dont need to care csiConfigMap is empty or not empty right

Reply:
Because if user will select "Choose existing KMS connection" radio button again, it will show him old connection details in the dropdown. Even though they do not exit any more, because dropdown is using csiConfigMap to display the connection details which still contains the older value.
Also, now if I create a new connection (without reloading the page)... it will give me "configmaps "csi-kms-connection-details" not found" error.

Reply:
ack